closed/src/java.base/share/classes/jdk/crypto/jniprovider/NativeCrypto.java

```diff
@@ -201,6 +202,8 @@ public void run() {
 
     private static final native long loadCrypto(boolean trace);
 
+    public static final native boolean isMD5Available();
```

Comment:
Suggest you generalize this as
```java
public static final native boolean isDigestAvailable(int algoIndex);
```

Reply:
This is just for `MD5`. There are other digests like `SHA` that are available. Unless you mean adding checks for them too and not failing if they are not present.

Reply:
Yes, I was suggesting that use of other digest algorithms should be conditional on their availability in the library present on the runtime system.

Reply:
Ok, so let's keep just for `MD5` for now and we'll add further functionality in future changes.

Reply:
In my opinion, keeping it as is represents a missed opportunity to avoid some future work, but I won't insist.